api/cpp/include/slint-stm32.h

```diff
@@ -178,8 +178,29 @@ inline __IO bool StmSlintPlatform::screen_ready = true;
 
 } // namespace slint::private_api
 
-inline void slint_stm32_init(const SlintPlatformConfiguration &config)
+inline int slint_stm32_init(const SlintPlatformConfiguration &config)
 {
+    /* Display initialization */
+    if (int err = BSP_LCD_InitEx(0, LCD_ORIENTATION_LANDSCAPE, LCD_PIXEL_FORMAT_RGB565,
+                                 config.size.width, config.size.height);
+        err != 0) {
+        return err;
+    }
+
+    BSP_LCD_DisplayOn(0);
+    BSP_LCD_SetActiveLayer(0, 0);
+
+    /* Touchscreen initialization */
+    TS_Init_t hTS {};
+    hTS.Width = config.size.width;
+    hTS.Height = config.size.height;
+    hTS.Orientation = TS_SWAP_XY;
```

Comment:
How/what to swap as well as the screen orientation differs between devices. Like for the H747 the way to avoid flicker is to use landscape, do software rotation, and swap the touch screen differently.